### PR TITLE
Bump nv-codecs for recent drivers, add support for NVDEC-enabled VAAP…

### PR DESCRIPTION
**Pull request by Mazotis**
…I driver and add related info in README

### CODE REVIEW

File: scripts/110-build-vaapi-nvidia [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+fi
+
+if [[ -d ${name} ]]; then
+    mkdir -p ${name}/build && cd ${name}/build
+    LDFLAGS="-L/opt/nvidia/lib/" LIBRARY_PATH="/opt/nvidia/lib" PKG_CONFIG_PATH="/usr/local/lib/pkgconfig" \
+    meson setup .. --prefix=/usr/local --libdir=lib && \
+    ninja install
+    if [[ $? -eq 0 && ! -f /usr/lib64/dri/nvidia-nvdec_drv_video.so ]]; then
+        ln -sf /usr/local/lib64/dri/nvidia_drv_video.so /usr/lib64/dri/nvidia-nvdec_drv_video.so
+    fi
```
Comment (marioroy) [inline scripts/110-build-vaapi-nvidia:38]: Now that passing `--libdir=lib` to meson setup, the file resides in lib dir. Update path.

```bash
    if [[ $? -eq 0 && ! -f /usr/lib/dri/nvidia-nvdec_drv_video.so ]]; then
        ln -sf /usr/local/lib/dri/nvidia_drv_video.so /usr/lib64/dri/nvidia-nvdec_drv_video.so
    fi
```

**Comment by marioroy** [comment]
One last thing. Some folks have the NVIDIA driver installed under /opt/nvidia but libs to be /opt/nvidia/lib64. So this is safety to work for both depending on how the NVIDIA driver installation installed libs to lib64 or lib.

```bash
if [[ -d ${name} ]]; then
    mkdir -p ${name}/build && cd ${name}/build
    LDFLAGS="-L/opt/nvidia/lib64 -L/opt/nvidia/lib" \
    LIBRARY_PATH="/opt/nvidia/lib64:/opt/nvidia/lib" \
    PKG_CONFIG_PATH="/usr/local/lib/pkgconfig" \
    meson setup .. --prefix=/usr/local --libdir=lib && \
    ninja install
    if [[ $? -eq 0 && ! -f /usr/lib/dri/nvidia-nvdec_drv_video.so ]]; then
        ln -sf /usr/local/lib/dri/nvidia_drv_video.so /usr/lib64/dri/nvidia-nvdec_drv_video.so
    fi
fi
```

File: scripts/110-build-vaapi-nvidia (hunk 1 above)
Comment (marioroy) [reply scripts/110-build-vaapi-nvidia:38]: Sorry, the if statement too; change to lib.
```bash
if [[ $? -eq 0 && ! -f /usr/lib/dri/nvidia-nvdec_drv_video.so ]]; then
```

**Comment by marioroy** [comment]
Hi @Mazotis.

The reason for minimum changes outside of `/usr/local` or `/opt` is safeguarding from `swupd` ridding of files (i.e. `swupd repair`). There's an ignore list for `swupd` and I have yet the time to investigate this.

So for now this is the reason for updating `LIBVA_DRIVERS_PATH` instead, in `~/.config/firefox.conf`. The README file mentions this in the content for `firefox.conf`.

There are a couple Firefox settings which are critical ensuring hardware acceleration. These are mentioned too.

The NVIDIA modules do not reside inside the boot image, under the boot partition. For that reason, `/etc/modprobe.d/enable-nvidia-modeset.conf` makes more sense, IMHO. If that isn't enough justification, there's another reason. Depending on how Clear Linux is installed (i.e. text-mode UI), it creates a single file `/etc/kernel/cmdline` containing kernel options on a single line.

Thanks,
Mario

**Comment by Mazotis** [comment]
Thanks. I did not know about the LIBVA_DRIVERS_PATH env variable - this removes the need for symlinking and seems much better. I always try to avoid /usr/lib or /usr/lib64 if possible as this indeed tends to create problems in the long run.

**Pull request merged**